tools/src/wyvern/tools/typedAST/core/declarations/ConstructDeclaration.java

```diff
@@ -0,0 +1,126 @@
+package wyvern.tools.typedAST.core.declarations;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import wyvern.target.corewyvernIL.FormalArg;
+import wyvern.target.corewyvernIL.decltype.DeclType;
+import wyvern.target.corewyvernIL.decltype.ValDeclType;
+import wyvern.target.corewyvernIL.expression.Variable;
+import wyvern.target.corewyvernIL.modules.TypedModuleSpec;
+import wyvern.target.corewyvernIL.support.GenContext;
+import wyvern.target.corewyvernIL.support.TopLevelContext;
+import wyvern.target.corewyvernIL.type.ValueType;
+import wyvern.tools.errors.FileLocation;
+import wyvern.tools.typedAST.abs.Declaration;
+import wyvern.tools.typedAST.core.binding.NameBinding;
+import wyvern.tools.typedAST.core.binding.NameBindingImpl;
+import wyvern.tools.typedAST.interfaces.CoreAST;
+import wyvern.tools.typedAST.interfaces.ExpressionAST;
+import wyvern.tools.typedAST.interfaces.TypedAST;
+import wyvern.tools.types.Type;
+import wyvern.tools.types.UnresolvedType;
+
+public class ConstructDeclaration extends Declaration implements CoreAST {
+    private ExpressionAST definition;
+    private NameBinding binding;
+    private List<NameBinding> argNames;
+
+    public ConstructDeclaration(String name, List<NameBinding> argNames, TypedAST definition, FileLocation location) {
+        this.definition = (ExpressionAST) definition;
+        binding = new NameBindingImpl(name, null);
+        this.argNames = argNames;
+        this.location = location;
+    }
+
+    // @Override
+    // public List<NameBinding> getArgBindings() {
+    //     return argNames;
+    // }
+
+    @Override
+    public Type getType() {
```

Comment:
binding.getType() is null.  Would be better to put code that generates the UnresolvedType (from below) here (or maybe in the constructor, in which case you could put it in binding.getType()) and then call this function down below.